opencti-platform/opencti-front/src/components/FilterIconButtonContentWithRedirection.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { FunctionComponent } from 'react';
+import { Link } from 'react-router-dom';
+import { Tooltip } from '@mui/material';
+import FilterIconButtonContentWithRedirectionAndDeprecatedInstancesSearch
+  from './FilterIconButtonContentWithRedirectionAndDeprecatedInstancesSearch';
+import { truncate } from '../utils/String';
+import { useFormatter } from './i18n';
+import { TriggerLine_node$data } from '../private/components/profile/triggers/__generated__/TriggerLine_node.graphql';
+
+interface FilterIconButtonContentWithRedirectionProps {
+  filter: { id: string, value: string },
+  resolvedInstanceFilters?: TriggerLine_node$data['resolved_instance_filters'];
+}
+
+interface ResolvedInstance {
+  readonly id: string,
+  readonly valid: boolean,
+  readonly value?: string,
+}
+
+const FilterIconButtonContentWithRedirection: FunctionComponent<FilterIconButtonContentWithRedirectionProps> = ({ filter, resolvedInstanceFilters }) => {
+  const { t } = useFormatter();
+  const displayedValue = filter.value && filter.value.length > 0
+    ? truncate(filter.value, 15)
+    : t('No label');
+
+  const redirectionWithResolvedInstances = () => {
+    const resolvedInstanceFiltersMap = new Map((resolvedInstanceFilters as readonly ResolvedInstance[]).map((n) => [
```

Comment:
Why do you need to change the type if the correct one comes from relay ?

Reply:
the type is not correct... for instance resolvedInstanceFilters can be null or undefined (because this attribute resolved_instance_filters is not mandatory in type Trigger, since this type includes both Live Triggers [which have filters and resolved_instance_filters] and Digest [which have no filters and no resolved_instance_filters])

Reply:
but I'm going to simplify this, by having only the nullity of the entire attribute to check